### ISSUE TITLE
ConnectionResetError on Python 3.8 for windows

### ISSUE THREAD

**Issue by erezsh** [prompt-toolkit/python-prompt-toolkit#1023]
Using the simplest feature (`prompt`) fails with async exception.

After running the prompt, I kept pressing `a`,`enter`.

I'm using Windows 10, `prompt_toolkit.__version__ = '2.0.10'`

```python
Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:21:23) [MSC v.1916 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from prompt_toolkit import prompt
>>>
>>> while 1:
...     user_input = prompt('>')
...     print(user_input)
...
>a
a
>a
a
>a
a

Unhandled exception in event loop:
  File "C:\Python38\lib\asyncio\proactor_events.py", line 768, in _loop_self_reading
    f.result()  # may raise
  File "C:\Python38\lib\asyncio\windows_events.py", line 808, in _poll
    value = callback(transferred, key, ov)
  File "C:\Python38\lib\asyncio\windows_events.py", line 457, in finish_recv
    raise ConnectionResetError(*exc.args)

Exception [WinError 995] The I/O operation has been aborted because of either a thread exit or an application request
Press ENTER to continue...
>a
a
>a
a
>a
a
>a
a

Unhandled exception in event loop:
  File "C:\Python38\lib\asyncio\proactor_events.py", line 768, in _loop_self_reading
    f.result()  # may raise
  File "C:\Python38\lib\asyncio\windows_events.py", line 808, in _poll
    value = callback(transferred, key, ov)
  File "C:\Python38\lib\asyncio\windows_events.py", line 457, in finish_recv
    raise ConnectionResetError(*exc.args)

Exception [WinError 995] The I/O operation has been aborted because of either a thread exit or an application request
Press ENTER to continue...
>a
a
>a
a
>                                                                                                                       
```

**Comment by erezsh** [comment]
Note: This happens in both x32 and x64, but much less often in x64.

**Comment by jonathanslenders** [comment]
Thanks for reporting @erezsh! 

I'm not sure what's going on, and wonder whether it's maybe even a bug in asyncio...

Can you see whether this is also the case for the ``SelectorEventLoop``?

Run the following snippet before your code:
```python
import asyncio
import selectors

selector = selectors.SelectSelector()
loop = asyncio.SelectorEventLoop(selector)
asyncio.set_event_loop(loop)
```

**Comment by erezsh** [comment]
That seems to cure it!

What's going on here? Is it safe to use in my code?

**Comment by jonathanslenders** [comment]
Python 3.8 made the ``ProactorEventLoop`` the default on Windows. The code above tells Python to use a ``SelectorEventLoop`` instead, which is similar to Python 3.7 and before.

@vstinner: Could this be a bug in the ``ProactorEventLoop``?

**Comment by vstinner** [comment]
I have no idea. If you think that it is a bug, write a producer and post it to bugs.python.org.

**Comment by jonathanslenders** [comment]
Issue opened at bugs.python.org: https://bugs.python.org/issue39010

**Comment by albaraa-elhag** [comment]
Getting the same error when I run ijupyter in terminal. 

```
Unhandled exception in event loop:
  File "c:\python38\lib\asyncio\proactor_events.py", line 768, in _loop_self_reading
    f.result()  # may raise
  File "c:\python38\lib\asyncio\windows_events.py", line 808, in _poll
    value = callback(transferred, key, ov)
  File "c:\python38\lib\asyncio\windows_events.py", line 457, in finish_recv
    raise ConnectionResetError(*exc.args)

Exception [WinError 995] The I/O operation has been aborted because of either a thread exit or an application request
Press ENTER to continue...
```

**Comment by musm** [comment]
Also getting this, do we have a workaround?

**Comment by jonathanslenders** [comment]
This was fixed in Python: https://github.com/python/cpython/pull/22017
Best is to upgrade to a Python version that includes the fix, like Python 3.8.7.

Otherwise, the workaround is to make sure a ``SelectorEventLoop`` is used by asyncio. See for instance how it's done here: https://github.com/xonsh/xonsh/issues/3430
